src/Payments/Integrations/Stripe.php

```diff
@@ -181,6 +182,53 @@ private function get_button_locale() {
 		return apply_filters( 'wc_stripe_payment_request_button_locale', substr( get_locale(), 0, 2 ) );
 	}
 
+	/**
+	 * Return the icons urls.
+	 *
+	 * @return array Array containing arrays of icons metadata.
+	 */
+	private function get_icons() {
+		$icons_src = [
+			[
+				'id'  => 'visa',
+				'url' => WC_STRIPE_PLUGIN_URL . '/assets/images/visa.svg',
+				'alt' => __( 'Visa', 'woo-gutenberg-products-block' ),
+			],
```

Comment:
I'm not sure why this has to be an array of arrays. I think it'd be better to do:

```js
$icons_src = [
    'visa' => [ 'url' => WC_STRIPE_PLUGIN_URL . '/assets/images/visa.svg', 'alt' => __( 'Visa', 'woo-gutenberg-products-block' ) ]
];
```

I know you give the reason:

> I went with an array of objects instead of an object since this will have arbitrary keys (name of cc).

...but I'm not sure why that matters.  Having the id as the index:

- ensures there's only ever ONE icon for each company. 
- makes it easier for extensions to filter and replace images for a given icon (if/when we add filtering in the future).

I may not understand your rationale so I'm open to more details there, but based on what I see I'd like to see the shape changed.

Reply:
this was the first case I went with and. I had the same reasons in mind, then I went on to write the types for it and I kinda felt that a key should probably be a constant name related to that data, not the part of the actual data, however, I don’t have any strong feelings about this and I'm happy to revert it.